### PR TITLE
Overriding a remoting type warning when a remote model has relations

### PR DESCRIPTION
**Pull request by angfal**
#### Related issues

- connect to #81

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by bajtos** [comment]
@slnode ok to test

**Comment by bajtos** [comment]
Hi @angfal, thank you for the pull request! I was on vacation for two weeks, I'll try to review your changes as soon as possible, hopefully by Friday.

File: lib/relations.js [hunk 1]
```diff
@@ -75,8 +75,7 @@ function RelationMixin() {
  * @property {Object} model Model object
  */
 RelationMixin.hasMany = function hasMany(modelTo, params) {
-  var def = RelationDefinition.hasMany(this, modelTo, params);
-  this.dataSource.adapter.resolve(this);
+  let def = RelationDefinition.hasMany(this, modelTo, params);
```
Comment (bajtos) [inline lib/relations.js]: Use `const` - see http://loopback.io/doc/en/contrib/style-guide.html#variable-declarations
  
Please fix other places in this file too.

File: test/models-resolve.test.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    clientApp = loopback({localRegistry: true});
+    remoteDs = helper.createRemoteDataSource(clientApp, serverApp);
+  });
+
+  before('create a child model', () => {
+    const db = helper.createMemoryDataSource(clientApp);
+    ChildModel = clientApp.registry.createModel({
+      name: 'ChildModel',
+    });
+    clientApp.model(ChildModel, {dataSource: db});
```
Comment (bajtos) [inline test/models-resolve.test.js]: I find it rather unusual to have a relation between a ChildModel that's attached to (persisted in) a local in-memory datasource, and a RemoteModel that's attached to a remote datasource on the server. Is that intentional? Did you perhaps mean to attach ChildModel to remoteDs too?

File: test/models-resolve.test.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      name: 'RemoteModelHasMany',
+      relations: {
+        children: {
+          type: 'hasMany',
+          model: 'ChildModel',
+        },
+      },
+    });
+    clientApp.model(RemoteModel, {dataSource: remoteDs});
+    assert(resolveSpy.withArgs(RemoteModel).calledOnce);
```
Comment (bajtos) [inline test/models-resolve.test.js]: Please use [Sinon assertions](http://sinonjs.org/releases/v4.1.3/assertions/).

`assert(<code returning true or false>)` produces failure messages that are difficult to troubleshoot - they only say that the test failed, but don't provide any additional insights about what went wrong.

I believe Sinon Asserts provide better error messages.
  
Please all fix other newly added tests in a similar way.

**Comment by bajtos** [comment]
@angfal As I have re-read my review comment above, it occurs to me that it may sound like I am rejecting your patch, which is not true. Your changes looks reasonable to me, it's just that I am not very familiar with this codebase and would like to better understand possible ramifications of this change, before I land it.

**Comment by angfal** [comment]
Thank you for feedback. I will fix and comment it during the weekend

**Comment by angfal** [comment]
It's wrong fix, I will send another. The second resolve execution allow to initiate additional remote methods for relations. The problem point only here:

https://github.com/strongloop/loopback-connector-remote/blob/906979a960fd0e4f53f49237d5e634f573d9a8e7/lib/remote-connector.js#L89

It defines type for the current model the second time. It should be moved to `connector.define()` method I think

**Comment by bajtos** [comment]
For posterity, here is the follow-up pull request: #85

@angfal in the future, please keep the same feature branch and pull request even if you revamp the code, so that all relevant discussion stays in a single place. Git history is easy to clean once the proposed changes are approved.

**Comment by angfal** [comment]
@bajtos Sorry, didn't think about the history. I'll fix it in the next time

**Pull request closed without merging**